flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputChannel.java

```diff
@@ -276,21 +277,36 @@ protected int getCurrentBackoff() {
      * @return <code>true</code>, iff the operation was successful. Otherwise, <code>false</code>.
      */
     protected boolean increaseBackoff() {
+        return increaseBackoff(0);
+    }
+
+    /**
+     * The remote task manager creates partition request listener and returns {@link
+     * PartitionNotFoundException} until the listener is timeout, so the backoff should add the
+     * timeout milliseconds if it exists.
+     *
+     * @param timeoutMS The timeout milliseconds that the partition request listener timeout
+     * @return <code>true</code>, iff the operation was successful. Otherwise, <code>false</code>.
+     */
+    protected boolean increaseBackoff(int timeoutMS) {
         // Backoff is disabled
         if (currentBackoff < 0) {
             return false;
         }
 
         // This is the first time backing off
         if (currentBackoff == 0) {
-            currentBackoff = initialBackoff;
+            currentBackoff = timeoutMS > 0 ? timeoutMS : initialBackoff;
 
             return true;
         }
 
         // Continue backing off
         else if (currentBackoff < maxBackoff) {
-            currentBackoff = Math.min(currentBackoff * 2, maxBackoff);
+            currentBackoff =
+                    timeoutMS > 0
+                            ? Math.min(currentBackoff + timeoutMS, maxBackoff)
+                            : Math.min(currentBackoff * 2, maxBackoff);
```

Comment:
Why do we have two different conditions here? Especially that with the default value for `timeoutMs` to be 10s, and `maxBackoff` also by default being 10s, both conditions return `maxBackoff` anyway? 

If so, maybe let's keep the original logic for the sake of simplicity?